book.md

```diff
@@ -1490,6 +1490,89 @@ now explore what we can do about these.
+Because the connections for delivering power (at the power supply, printed circuit board, on chip packages as well as on the silicon dices) also contain resistance, the variation of electrical current in the chip’s power supply also results in variations in the power supply voltage. Again, this can be observed easily if the attacker has physical access to the device.
+If an attacker has access to data acquisition equipment that can record the current/voltage/power patterns, the attacker can record the “power signature” for different crypto operations, including the power signature using different data inputs. By applying analysis techniques like differential power analysis, the attacker could extract the information being processed.
+One additional form of side-channel leakage is electro-magnetic radiation. Because the processor’s clock frequency is usually in radio frequency (RF) range, the wires and tracks on the PCB becomes small antenna and the ripples in the processor’s voltage/current results in radio frequency signals. Although the RF power radiated can be tiny, it still means that an attacker can observe the side-channel leakage if he/she is in close proximity from the device and has the right equipment to amplify and record the RF power signature. However, the risk of such attack can be reduced by reducing the radiation energy level using:
+- Shielding around the device, including ground plate on the circuit board.
+- Coupling capacitors on power supply tracks on the printed circuit board.
+Generally, such attack requires knowledge of radio circuit techniques and the result can be affected by other factors. For example, in normal environments there are many other source of RF noises that affects the accuracy of signal measurement. In a “noisy” environment, the RF signals from various wireless communication gadgets nearby might drown out the signals from the device being monitored.
+
+### Side channel leakage at instruction level
+Instruction executions can result in various forms of side channel leakage:
+Cycle timing resulting from conditional branch – A code sequence containing a conditional branch could result in observable side channel leakage. For example, if the power signatures of several code segments are easily recognizable (process A, B and C in the following diagram), it is possible to detect if the conditional branch was taken or not.
+
+![Alt text](./side_channel_leakage_figure_1_branch.svg)
+<img src="./img/side_channel_leakage_figure_1_branch.svg">
+
+Cycle timing resulting from specific data value – Some instructions execution cycle can be dependent on the value of input data, resulting in timing side-channel leakage. E.g., the integer divide instruction in Arm Cortex-M processors.
```

Comment:
s/from specific data value/from a specific data value/?

Reply:
I changed it to "from specific data values". It is possible to have leakage based on multiple data values.